data_override/include/data_override.inc

```diff
@@ -579,54 +583,96 @@ subroutine read_table_yaml(data_table)
       call get_block_ids(file_id, "data_table", entry_id)
 
       do i = 1, nentries
-         call get_value_from_key(file_id, entry_id(i), "gridname", data_table(i)%gridname)
+         call get_value_from_key(file_id, entry_id(i), "grid_name", data_table(i)%gridname)
          call check_for_valid_gridname(data_table(i)%gridname)
-         call get_value_from_key(file_id, entry_id(i), "fieldname_code", data_table(i)%fieldname_code)
+         call get_value_from_key(file_id, entry_id(i), "fieldname_in_model", data_table(i)%fieldname_code)
+
+         mentries = get_num_blocks(file_id, "override_file", parent_block_id=entry_id(i))
+         if(mentries.gt.1) call mpp_error(FATAL, "Too many override_file blocks in data table. "//&
+              "Check your data_table.yaml entry for field:"//trim(data_table(i)%gridname)//":"//&
+              trim(data_table(i)%fieldname_code))
+         call get_block_ids(file_id, "override_file", sub_block_id, parent_block_id=entry_id(i))
+
+         data_table(i)%file_name = ""
+         call get_value_from_key(file_id, sub_block_id(1), "file_name", data_table(i)%file_name, &
+              & is_optional=.true.)
 
          data_table(i)%fieldname_file = ""
-         call get_value_from_key(file_id, entry_id(i), "fieldname_file", data_table(i)%fieldname_file, &
-           & is_optional=.true.)
+         call get_value_from_key(file_id, sub_block_id(1), "fieldname_in_file", data_table(i)%fieldname_file, &
+              & is_optional=.true.)
+
+         data_table(i)%interpol_method = "none"
+         call get_value_from_key(file_id, sub_block_id(1), "interp_method", data_table(i)%interpol_method, &
+              & is_optional=.true.)
```

Comment:
is_optional should be removed now, because you are overriding from a file so the interpol_method is required